src/replication.c

```diff
@@ -1831,7 +1843,7 @@ redisDb *disklessLoadInitTempDb(void) {
 /* Helper function for readSyncBulkPayload() to discard our tempDb
  * when the loading succeeded or failed. */
 void disklessLoadDiscardTempDb(redisDb *tempDb) {
-    discardTempDb(tempDb, replicationEmptyDbCallback);
+    discardTempDb(tempDb, NULL);
```

Comment:
Looks like discardTempDb became async only some time ago and callback won't be called anyway.

Reply:
discardTempDb() works async and it never calls `replicationEmptyDbCallback`. It passes current db to bio thread to be destroyed in the background. As destroying is not blocking, callback is unused.

Reply:
from https://github.com/redis/redis/pull/9323/files#diff-7adb516f69cdcafa3c44249b43cc789b564182dbfce54da02df047c20afe5718R488

Reply:
so if it's always async, let's drop the callback argument form the function.

Reply:
ohh. you did.